covariance/covarianceBase.h

```diff
@@ -506,16 +495,45 @@ class covarianceBase {
   /// Throw matrix that is being used in the fit, much faster as TMatrixDSym cache miss
   double **throwMatrixCholDecomp;
 
-  /// @brief HW: Truly Adaptive Stuff
-  void initialiseNewAdaptiveChain();
-  // TMatrixD* getMatrixSqrt(TMatrixDSym* inputMatrix);
-  // double calculateSubmodality(TMatrixD* sqrtVectorCov, TMatrixDSym* throwCov);
-  /// HW: Do we use adaptive MCMC or not
+  /// @brief sets default values for adaptive MCMC parameters
+  void setAdaptionDefaults();
+
+  /// @brief sets adaptive block matrix
+  /// @param block_indices Values for sub-matrix blocks
+  void setAdaptiveBlocks(std::vector<std::vector<int>> block_indices);
+
+  /// @brief sets throw matrix from a file
+  /// @param matrix_file_name name of file matrix lives in
+  /// @param matrix_name name of matrix in file
+  /// @param means_name name of means vec in file
+  void setThrowMatrixFromFile(std::string matrix_file_name, std::string matrix_name, std::string means_name);
+
+  /// @brief Method to update adaptive MCMC
+  void updateAdaptiveCovariance();
+  /// @brief method to create new throw matrix
+  void createNewAdaptiveCovariance();
+
+  // Are we using AMCMC?
   bool use_adaptive;
+  // Total number of steps chain has performed
   int total_steps;
-  int lower_adapt, upper_adapt; //Thresholds for when to turn on/off adaptive MCMC
-  // TMatrixD* covSqrt;
+  // When do we start throwing?
+  int start_adaptive_throw;
+  //Thresholds for when to turn on/off updating adaptive MCMC
+  int start_adaptive_update;
+  // When do we stop update the adaptive matrix?
+  int end_adaptive_update;
+  // Steps between changing throw matrix
+  int adaptive_update_step;
+
+
+  // Indices for block-matrix adaption
+  std::vector<int> adapt_block_matrix_indices;
+  // Size of blocks for adaption
+  std::vector<int> adapt_block_sizes;
```

Comment:
nitpicky but if you replace // with /// above variable then this is recongised by Doxygen. Would be nice to have this in doxy docueamtnion